src/smt/set_defaults.cpp

```diff
@@ -92,12 +92,14 @@ void setDefaults(LogicInfo& logic, bool isInternalSubsolver)
     }
   }
   if (options::unsatCoresNew()
+      && options::unsatCoresMode() != options::UnsatCoresMode::FULL_PROOF
       && ((options::produceProofs() && options::produceProofs.wasSetByUser())
           || (options::checkProofs() && options::checkProofs.wasSetByUser())
           || (options::dumpProofs() && options::dumpProofs.wasSetByUser())))
   {
-    AlwaysAssert(false) << "Can't properly produce proofs and have the new "
-                           "unsat cores simultaneously.\n";
+    AlwaysAssert(false)
+        << "Can't simultaneously produce proofs and not have the new "
+           "unsat cores in full proof mode .\n";
   }
   if (options::checkProofs() || options::unsatCoresNew()
```

Comment:
You should also enable proofs if unsat core mode was set to full proof.

Reply:
We need to set produceProofs for all unsat core modes, so I think this is already taken care of by your previous suggestion.